index.html

```diff
@@ -5,45 +5,510 @@
+
+
+function compareStrings (a,b) {
+  a = a.toLowerCase();
+  b = b.toLowerCase();
+  return (a < b) ? -1 : (a > b) ? 1: 0;
+} 
+
+
+    if (options.sorted) {
+      if (options.sorted === 'desc-income') {
+        console.log("sorted")
+        data.sort(function (a,b) {
+          return d3.descending(a.avg_family_income, b.avg_family_income); });
+      }  
+    } else if (options.filtered) {
+      if (options.filtered === 'filter-high-poverty') {
+        data = data.filter(function (a) { return a.poverty_rate > 20; });
+      } else if (options.filtered === 'filter-high-income') {
+      data = data.filter(function (a) { return a.avg_family_income > 99000; });
+    }   
+  }
+    else { 
+        data.sort(function (a,b) { return compareStrings(a.neighborhood, b.neighborhood); });
+      }
+
+ var rows = tbody.selectAll("tr")
+    .data(data)
+    .enter().append("tr")
+
+  //rows.append("tr")
+  
+
+/*
+ var rows = tbody.selectAll("tr")
+  .data(dc_info)
+  .enter()
+  .append("tr")
+  if (options.sorted) {
+    if (options.sorted === 'desc-income') {
+      rows.sort(function (a,b) { return d3.descending (a.avg_family_income, b.avg_family_income); });
+    } else if (options.sorted === 'desc-poverty') {
+      rows.sort(function (a,b) { return d3.descending (a.poverty_rate, b.poverty_rate); });
+    } 
+  } else if (options.filtered) {
+      if (options.filtered === 'filter-high-poverty') {
+        rows.filter(function (a) { return a.poverty_rate > 20; });
+      } else if (options.filtered === 'filter-high-income') {
+      rows.filter(function (a) { return a.avg_family_income > 99000; });
+    }   
+  } else {
+      rows.sort(function (a,b) { return compareStrings (a.neighborhood, b.neighborhood);});
+    }
+
+  */
+
+
+  rows.append("td")
+    .html(function(d) { return d.neighborhood});
+
+  rows.append("td")
+    .html(function(d) { return "$" + d3.format(",")(d.avg_family_income)});
+
+  rows.append("td")
+    .append("div")
+    .classed("bar", true)
+    .style("width", function(d) { return ((d.avg_family_income / MAX_INCOME) * 100 + '%')});
+
+  rows.append("td")
+    .html(function(d) { return d.poverty_rate + "%"});
+
+  rows.append("td")
+    .append("div")
+    .classed("bar", true)
+    .style("width", function(d) { return ((d.poverty_rate / MAX_POVERTY) * 100 + '%')})
+
+
+
+/*
+  dc_info.select("#sort-desc-income")
+    .data(dc_info)
+    .on("click", function(){
+    rows.sort(function(a, b) { return b.avg_family_income - a.avg_family_income; });
+    });
+    */
+};
+
+
+updateTable();
+
+
+
+
+d3.select("#sort-desc-income")
+  .on("click", function() {
+    if (options.sorted === 'desc-income') {
+    options.sorted = false;
+  } else {
+    options.sorted = 'desc-income';
+  }
+    updateTable();
+    d3.select('#sort-desc-income').classed('active', true);
+    d3.select('#sort-desc-poverty').classed('active', false);
+    d3.select('#filter-poverty').classed('active', false);
```

Comment:
And note that if we're changing the active class on these other buttons, we should also be setting options.filtered to false so that our data and our browser show the same state.